common/build.gradle

```diff
@@ -60,3 +78,132 @@ dependencies {
+}
+
+// Task to generate javadoc.jar
+task javadocJar(type: Jar, dependsOn: javadoc) {
+    classifier = 'javadoc'
+    from javadoc.destinationDir
+    destinationDir = reporting.file("$project.buildDir/outputs/jar/")
+}
+
+task sourcesJar(type: Jar) {
+    from android.sourceSets.main.java.srcDirs
+    classifier = 'sources'
+    destinationDir = reporting.file("$project.buildDir/outputs/jar/")
+}
+
+task createPom  {
+    pom {
+        project {
+            groupId 'com.microsoft.identity.common'
+            artifactId 'common'
+            version = project.version
+            packaging 'aar'
+            name 'common'
+
+            description 'This library contains code shared between the Active Directory Authentication Library (ADAL) for Android and the Microsoft Authentication Library (MSAL) for Android. This library includes only internal classes and is NOT part of the public API'
+            url 'https://github.com/AzureAD/microsoft-authentication-library-common-for-android'
+
+            developers {
+                developer {
+                    id 'microsoft'
+                    name 'Microsoft'
+                }
+            }
+
+            licenses {
+                license {
+                    name 'MIT License'
+                }
+            }
+            inceptionYear '2014'
+
+            properties {
+                branch 'master'
+                commonVersion = project.version
+            }
+
+            scm {
+                url "https://github.com/AzureAD/microsoft-authentication-library-common-for-android/tree/master"
+            }
+        }
+    }.writeTo("${archivesBaseName}-${version}.pom")
+}
+
+// For publishing to the remote maven repo.
+publishing {
+    publications {
+        adal(MavenPublication) {
+            groupId 'com.microsoft.identity'
+            artifactId 'common'
+            //Edit the 'version' here for VSTS RC build
+            version = project.version
+
+            pom.withXml {
+                def dependenciesNode = asNode().appendNode('dependencies')
+
+                //Iterate over the compile dependencies (we don't want the test ones), adding a <dependency> node for each
+                configurations.compile.allDependencies.each {
+                    if (it.group != null && it.name != null) {
+                        def dependencyNode = dependenciesNode.appendNode('dependency')
+                        dependencyNode.appendNode('groupId', it.group)
+                        dependencyNode.appendNode('artifactId', it.name)
+                        dependencyNode.appendNode('version', it.version)
+                    }
+                }
+            }
+
+            artifact(sourcesJar)
+            artifact(javadocJar)
+            artifact("$buildDir/outputs/aar/common-${project.version}.aar")
+        }
+    }
+
+    // Repositories to which Gradle can publish artifacts
+    repositories {
+        maven {
+            name "vsts-maven-adal-android"
```

Comment:
Verify the target, URL, and such here -- this appears to be ADAL's config.

Reply:
We upload all the MSAL, ad-accounts, common and ADAL to the same VSTS AndroidADAL url. Verified by publishing common here : [link](https://identitydivision.visualstudio.com/IDDP/DevEx-Client-SDK/_packaging?feed=AndroidADAL&package=com.microsoft.identity%3Acommon&version=0.0.1&protocolType=maven&_a=package)